### PR TITLE
Add a pre-commit hook for developers

### PR DESCRIPTION
**Pull request by IRDonch**
### Summary
It will help to remind people to format their code before committing.

This also establishes `dev` as a directory for developer-oriented scripts.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by IRDonch** [comment]
CC @kirill-sizov @yasakova-anastasia

File: dev/pre-commit.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# SPDX-License-Identifier: MIT
+
+"""
+This script can be used as a pre-commit hook when contributing to this project.
+It runs a subset of CI checks that is sufficiently fast to not be annoying when run
+for every commit.
+
+To install, make sure isort and black are installed on your system, then run:
+
+$ ln -sv ../../dev/pre-commit.py ./.git/hooks/pre-commit
```
Comment (zhiltsov-max) [inline dev/pre-commit.py:14]: Probably, it would be good to mention this in Developer guide.

Comment (IRDonch) [reply dev/pre-commit.py:14]: I added a mention in `CONTRIBUTING.md`, since that's where the coding style is described.

**Comment by IRDonch** [comment]
I updated the hook so that only changed files are scanned. This makes it much faster.

**Review by zhiltsov-max — APPROVED** [review]

**Pull request merged**